src/Dialect/ONNX/ONNXOps/Tensor/Constant.cpp

```diff
@@ -70,3 +70,27 @@ LogicalResult ONNXConstantOp::inferShapes(
 namespace onnx_mlir {
 template struct ONNXNonSpecificOpShapeHelper<ONNXConstantOp>;
 } // namespace onnx_mlir
+
+//===----------------------------------------------------------------------===//
+// Folder
+//===----------------------------------------------------------------------===//
+
+OpFoldResult ONNXConstantOp::fold(FoldAdaptor adaptor) {
+  if (getSparseValueAttr())
+    return getSparseValueAttr();
+  if (getValueAttr())
+    return getValueAttr();
+  else if (getValueFloatAttr())
+    return getValueFloatAttr();
+  else if (getValueIntAttr())
+    return getValueIntAttr();
+  else if (getValueIntsAttr())
+    return getValueIntsAttr();
+  else if (getValueFloatsAttr())
+    return getValueFloatsAttr();
+  else {
+    assert(getValueStringAttr() &&
+           "ONNXConstantOp does not have a valid attribute");
+    return getValueStringAttr();
+  }
```

Comment:
we should also check getValueStringsAttr(), right?

sorry for trying to rewrite the whole method but I'd prefer if we write it as:
```
if (auto sparseValue = getSparseValueAttr()) return sparseValue;
if (auto value = getSparseValueAttr()) return value;
...
if (auto valueStrings = getValueStringsAttr()) return valueStrings;
llvm_unreachable("ONNXConstantOp does not have a valid attribute");
```
so it's consistent and symmetrical for ease of reading, and we don't look up any attribute twice